src/main/java/gov/nih/nci/bento/controller/GraphQLController.java

```diff
@@ -41,32 +35,51 @@
 
 import javax.annotation.PostConstruct;
 import javax.servlet.http.HttpServletResponse;
-import java.io.File;
 import java.io.IOException;
-import java.nio.file.Files;
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 @RestController
-@DependsOn({"neo4jDataFetcher"})
+@DependsOn({"privateNeo4jDataFetcher", "publicNeo4jDataFetcher"})
 public class GraphQLController {
 
 	private static final Logger logger = LogManager.getLogger(GraphQLController.class);
 
 	@Autowired
 	private ConfigurationDAO config;
 	@Autowired
-	private Neo4jDataFetcher dataFetcherInterceptor;
+	private PrivateNeo4jDataFetcher privateNeo4jDataFetcher;
 	@Autowired
-	private DataFetcher esFilterDataFetcher;
-
+	private PublicNeo4jDataFetcher publicNeo4JDataFetcher;
+	@Autowired
+	@Qualifier("privateESDataFetcher")
+	private AbstractESDataFetcher privateESDataFetcher;
+	@Autowired
+	@Qualifier("publicESDataFetcher")
+	private AbstractESDataFetcher publicESDataFetcher;
 
 	private Gson gson = new GsonBuilder().serializeNulls().create();
-	private GraphQL graphql;
+	private GraphQL privateGraphQL;
+	private GraphQL publicGraphQL;
+
+	@PostConstruct
+	public void initGraphQL() throws IOException {
```

Comment:
This is an optional suggestion. GraphQLController should minimize general business logic. We could declare and allocate privateGraphQL and publicGraphQL in BuildBentoGraphQL instead.